### DISCUSSION TITLE
logit vs. log probability in powerset to multilabel conversion

### DISCUSSION THREAD

**Discussion by lukasstorck** [pyannote/pyannote-audio#1703] [Q&A] [answered]
In the [`Powerset.to_multilabel(...)` function](https://github.com/pyannote/pyannote-audio/blob/704f87414a57173938ba546115bbd71de02da82b/pyannote/audio/utils/powerset.py#L112) with `soft=True` the docstring asks for "logit" values:

```python
def to_multilabel(self, powerset: torch.Tensor, soft: bool = False) -> torch.Tensor:
    """Convert predictions from powerset to multi-label

    Parameter
    ---------
    powerset : (batch_size, num_frames, num_powerset_classes) torch.Tensor
        Soft predictions in "powerset" space.
    soft : bool, optional
        Return soft multi-label predictions. Defaults to False (i.e. hard predictions)
        Assumes that `powerset` are "logits" (not "probabilities").

    Returns
    -------
    multi_label : (batch_size, num_frames, num_classes) torch.Tensor
        Predictions in "multi-label" space.
    """
```

As I understand, [logits](https://en.wikipedia.org/wiki/Logit) (ranging from -inf to +inf) are the values in the model before applying the activation function. In case of the `pyannote/segmentation-3.0` model, the output is in the powerset representation and the activation function of the last layer is "LogSoftmax", which results in [log probabilities](https://en.wikipedia.org/wiki/Log_probability) ranging from -inf to 0. These values are the values used by the powerset to multilabel conversion and not the values before the activation function.

I was confused by the use of the term "logit" in the docstring. Isn't "log probabilities" the correct term here? Thank you :D

**Comment by hbredin** [comment, marked as answer]
You are right. PR to fix the docstring very welcome!
